analyzers/src/SonarAnalyzer.CFG/LiveVariableAnalysis/LiveVariableAnalysisBase.cs

```diff
@@ -20,95 +20,94 @@
+    protected readonly ISymbol originalDeclaration;
+    protected readonly CancellationToken Cancel;
+    private readonly IDictionary<TBlock, HashSet<ISymbol>> blockLiveOut = new Dictionary<TBlock, HashSet<ISymbol>>();
+    private readonly IDictionary<TBlock, HashSet<ISymbol>> blockLiveIn = new Dictionary<TBlock, HashSet<ISymbol>>();
+    private readonly HashSet<ISymbol> captured = new();
 
-        public abstract bool IsLocal(ISymbol symbol);
-        protected abstract TBlock ExitBlock { get; }
-        protected abstract State ProcessBlock(TBlock block);
-        protected abstract IEnumerable<TBlock> ReversedBlocks();
-        protected abstract IEnumerable<TBlock> Successors(TBlock block);
-        protected abstract IEnumerable<TBlock> Predecessors(TBlock block);
+    public abstract bool IsLocal(ISymbol symbol);
+    protected abstract TBlock ExitBlock { get; }
+    protected abstract State ProcessBlock(TBlock block);
+    protected abstract IEnumerable<TBlock> ReversedBlocks();
+    protected abstract IEnumerable<TBlock> Successors(TBlock block);
+    protected abstract IEnumerable<TBlock> Predecessors(TBlock block);
 
-        public TCfg Cfg { get; }
-        public IReadOnlyCollection<ISymbol> CapturedVariables => captured;
+    public TCfg Cfg { get; }
+    public IReadOnlyCollection<ISymbol> CapturedVariables => captured;
 
-        protected LiveVariableAnalysisBase(TCfg cfg, ISymbol originalDeclaration, CancellationToken cancel)
-        {
-            Cfg = cfg;
-            this.originalDeclaration = originalDeclaration;
-            this.Cancel = cancel;
-        }
+    protected LiveVariableAnalysisBase(TCfg cfg, ISymbol originalDeclaration, CancellationToken cancel)
+    {
+        Cfg = cfg;
+        this.originalDeclaration = originalDeclaration;
+        this.Cancel = cancel;
+    }
 
-        /// <summary>
-        /// LiveIn variables are alive when entering block. They are read inside the block or any of it's successors.
-        /// </summary>
-        public IReadOnlyCollection<ISymbol> LiveIn(TBlock block) =>
-            blockLiveIn[block].Except(captured).ToImmutableArray();
+    /// <summary>
+    /// LiveIn variables are alive when entering block. They are read inside the block or any of it's successors.
+    /// </summary>
+    public IReadOnlyCollection<ISymbol> LiveIn(TBlock block) =>
+        blockLiveIn[block].Except(captured).ToImmutableArray();
 
-        /// <summary>
-        /// LiveOut variables are alive when exiting block. They are read in any of it's successors.
-        /// </summary>
-        public IReadOnlyCollection<ISymbol> LiveOut(TBlock block) =>
-            blockLiveOut[block].Except(captured).ToImmutableArray();
+    /// <summary>
+    /// LiveOut variables are alive when exiting block. They are read in any of it's successors.
+    /// </summary>
+    public IReadOnlyCollection<ISymbol> LiveOut(TBlock block) =>
+        blockLiveOut[block].Except(captured).ToImmutableArray();
```

Comment:
`IReadOnlyCollection` and `ToImmutableArray` are overkills. All usages do some linq or hashset with it => return `IEnumerable<ISymbol>`, without alternation